src/lib/coda_base/transaction_logic.ml

```diff
@@ -472,6 +488,12 @@ module Make (L : Ledger_intf) : S with type ledger := L.t = struct
           (Amount.of_fee (User_command.Payload.fee payload))
       in
       let%bind () = validate_nonces (Account.Nonce.succ nonce) account.nonce in
+      let%bind () =
+        (* TODO: Put actual value here *)
+        let current_global_slot = Global_slot.zero in
+        validate_time ~valid_until:payload.common.valid_until
+          ~current_global_slot
```

Comment:
Why do need to check against the current global slot when we undo a user command?
Shouldn't we check against the slot number when we apply the user command?